### PR TITLE
Fix structured_outputs handling and tool normalization in vLLM backend

### PR DESCRIPTION
**Pull request by ehofm**
## Summary
This PR adds support for JSON Schema-based structured generation in the vLLM backend and improves the handling of empty tool lists in the `GRPOTrainer`.

### Key Features:
- **JSON Schema Support**: Enables passing full JSON schemas (e.g., from Pydantic models) through `generation_kwargs` when using vLLM-serve.
- **Robust Parameter Handling**: `trl/scripts/vllm_serve.py` now automatically converts dictionary-based `structured_outputs` into the required `StructuredOutputsParams` objects.
- **Empty Tool Normalization**: Updated `VLLMClient` and `GRPOTrainer` to treat empty tool lists as `None`, preventing `NotImplementedError` when tools are initialized but not used.

### Using JSON Schemas (New)
You can now use Pydantic models to define structured outputs for vLLM-backed generation:
```
from pydantic import BaseModel
from trl import GRPOConfig, GRPOTrainer

# 1. Define your schema as a Pydantic model
class ReasoningResponse(BaseModel):
    reasoning: str
    answer: float

# 2. Convert to JSON schema dictionary
my_schema = ReasoningResponse.model_json_schema()

# 3. Pass the schema to GRPOConfig via generation_kwargs
training_args = GRPOConfig(
    output_dir="./output",
    use_vllm=True,
    generation_kwargs={
        "structured_outputs": {"json": my_schema}
    }
)

trainer = GRPOTrainer(
    model="...",
    args=training_args,
    train_dataset=dataset,
    reward_funcs=[...]
)
```

Fixes # [5154](https://github.com/huggingface/trl/issues/5154)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qgallouedec** [comment]
@codex review

File: trl/generation/vllm_generation.py [hunk 1]
```diff
@@ -647,8 +660,8 @@ def generate(self, prompts: list, num_generations: int, profiler: ProfilingConte
                     "max_tokens": max_completion_length,
                     "logprobs": 0,  # enable returning log probabilities; 0 means for the sampled tokens only
                 }
-                generation_kwargs[structured_outputs_key] = structured_outputs
                 generation_kwargs.update(self.generation_kwargs)
+                generation_kwargs[structured_outputs_key] = structured_outputs
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/generation/vllm_generation.py:665]: does 

```python
                if structured_outputs is not None:
                    generation_kwargs[structured_outputs_key] = structured_outputs
```

instead solves this issue?

**Comment by qgallouedec** [comment]
@codex review

**Comment by qgallouedec** [comment]
@codex review
